metaconfig-core/shared/src/main/scala/metaconfig/Conf.scala

```diff
@@ -24,7 +24,7 @@ sealed abstract class Conf extends Product with Serializable {
   def as[T](implicit ev: ConfDecoder[T]): Configured[T] =
     ev.read(this)
   def getSettingOrElse[T](setting: Setting, default: T)(
-      implicit ev: ConfDecoder[T]
+      implicit ev: ConfDecoderFactory[T]
```

Comment:
Breaking changes in the public Metaconfig API like in this diff here require a bump in the Scalafix major version. Custom Scalafix rules use this API to read configuration. For better or worse, I would like to avoid the situation where custom Scalafix rules that were compiled against Scalafix v0.9.14 stop working with Scalafix v0.9.15.

Is there a way to introduce `ConfDecoderFactory` without making it a superclass of `ConfDecoder`?

Reply:
I am BTW open to merge breaking changes in the `metaconfig.internal` package to unblock improvements in Scalafmt.

Reply:
I'm OK with binary breaking changes in only 2.11 since Scalafix rules compile only against 2.12.

Adding a new abstract method to `ConfDecoder[A]` is binary breaking for clients that do `new ConfDecoder[A] { ... }`.

It's possible to enable MiMa to verify automatically what changes are binary compatible. So far this project hasn't had large API changes so I haven't worried about it